src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/Validators/ValidationBehavior.shared.cs

```diff
@@ -143,45 +146,45 @@ public object Value
 		/// <summary>
 		/// Allows the user to override the property that will be used as the value to validate. This is a bindable property.
 		/// </summary>
-		public string ValuePropertyName
+		public string? ValuePropertyName
 		{
-			get => (string)GetValue(ValuePropertyNameProperty);
+			get => (string?)GetValue(ValuePropertyNameProperty);
 			set => SetValue(ValuePropertyNameProperty, value);
 		}
 
 		/// <summary>
 		/// Allows the user to provide a custom <see cref="ICommand"/> that handles forcing validation. This is a bindable property.
 		/// </summary>
-		public ICommand ForceValidateCommand
+		public ICommand? ForceValidateCommand
 		{
-			get => (ICommand)GetValue(ForceValidateCommandProperty);
+			get => (ICommand?)GetValue(ForceValidateCommandProperty);
 			set => SetValue(ForceValidateCommandProperty, value);
 		}
 
 		protected virtual string DefaultValuePropertyName => Entry.TextProperty.PropertyName;
 
-		protected virtual ICommand DefaultForceValidateCommand => new Command(ForceValidate);
+		protected virtual ICommand DefaultForceValidateCommand { get; }
 
 		/// <summary>
 		/// Forces the behavior to make a validation pass.
 		/// </summary>
-		public void ForceValidate() => _ = UpdateStateAsync(true);
+		public ValueTask ForceValidate() => UpdateStateAsync(true);
 
 		internal ValueTask ValidateNestedAsync(CancellationToken token) => UpdateStateAsync(true, token);
 
-		protected virtual object Decorate(object value) => value;
+		protected virtual object? Decorate(object? value) => value;
 
-		protected abstract ValueTask<bool> ValidateAsync(object value, CancellationToken token);
+		protected abstract ValueTask<bool> ValidateAsync(object? value, CancellationToken token);
 
-		protected override void OnAttachedTo(VisualElement bindable)
+		protected override async void OnAttachedTo(VisualElement bindable)
 		{
 			base.OnAttachedTo(bindable);
 
 			isAttaching = true;
 			currentStatus = ValidationFlags.ValidateOnAttaching;
 
 			OnValuePropertyNamePropertyChanged();
-			_ = UpdateStateAsync(false);
+			await UpdateStateAsync(false);
```

Comment:
it potentially will break the logic.

**isAttaching** should become **false** as soon as possible. Now it will have to await the UpdateStateAsync's finish.

We do not need to await that operation here.

Reply:
When is the the behavior officially "attached"?

If the behavior isn't "attached" until after `UpdateStateAsync` has completed, then we need to `await UpdateStateAsync()`.

Reply:
UpdateStateAsync runs validation check (changes IsValid property value and applies styles)

Unfortunately, we cannot control when the behavior is attached, cuz it's done by Xamarin.Forms that provides **OnAttachedTo** method notifying us that the behavior is already attached.

Reply:
I dug deeper into this and and the good news is that there are no breaking changes.

1. `UpdateStateAsync` uses `ValueTask` which, by definition runs synchronously the majority of the time
2. The only behavior that actually awaits the ValueTask is the `MultiValidationBehavior`, and I confirmed that it continues to work as expected in the sample app

Reply:
@brminnick ValidationBehavior can be extended by users for implementing their own validations (which can be async).